### ISSUE TITLE
[FORMATTING] Union clause messes up the indentation

### ISSUE THREAD

**Issue by 1nfected** [sql-formatter-org/sql-formatter#383]
When a sub-query contains a union clause, the indentation for the union clause and the query after, is incorrect, i.e. it is not indented at all.

**Expected Output**

```sql
SELECT ColA
  FROM MyTable
 WHERE ColA IN (
           SELECT ColA
             FROM MyTable
            WHERE ColA IS NOT NULL
        UNION ALL
           SELECT ColB
             FROM AnotherTable
            WHERE ColB IS NOT NULL
       )
```

**Actual Output**

```sql
   SELECT ColA
     FROM MyTable
    WHERE ColA IN (
             SELECT ColA
               FROM MyTable
              WHERE ColA IS NOT NULL
UNION ALL
   SELECT ColB
     FROM AnotherTable
    WHERE ColB IS NOT NULL
)
```

As you can see, the query after `UNION ALL` clause is not indented. This is also reproducible with a CTE query.

On a side note, I would also like to suggest if it would be possible to add new lines before and after the `UNION` keyword. So the expected out would be something like:

```sql
SELECT ColA
  FROM MyTable
 WHERE ColA IN (
           SELECT ColA
             FROM MyTable
            WHERE ColA IS NOT NULL

        UNION ALL

           SELECT ColB
             FROM AnotherTable
            WHERE ColB IS NOT NULL
       )
```
This helps the user to spot the union clause more easily.

**Usage**

I'm using the vs-code extension, v1.1.1

**Comment by nene** [comment]
That is indeed a bug.

Should add more tests for the `indentStyle: tabular*` modes to avoid such mishaps in the future.

**Comment by nene** [comment]
Released a fix in 9.0.1
